python/pyarrow/table.pxi

```diff
@@ -2387,3 +2429,60 @@ def _from_pydict(cls, mapping, schema, metadata):
         return cls.from_arrays(arrays, schema=schema, metadata=metadata)
     else:
         raise TypeError('Schema must be an instance of pyarrow.Schema')
+
+
+class TableGroupBy:
+    """
+    A grouping of columns in a table on which to perform aggregations.
+    """
+
+    def __init__(self, table, keys):
+        if isinstance(keys, str):
+            keys = [keys]
+
+        self._table = table
+        self.keys = keys
+
+    def aggregate(self, aggregations):
+        """
+        Perform an aggregation over the grouped columns of the table.
+
+        Parameters
+        ----------
+        aggregations : list[tuple(str, str)] or\
+                       list[tuple(str, str, FunctionOptions)]
```

Comment:
```suggestion
        aggregations : list[tuple(str, str)] or \
list[tuple(str, str, FunctionOptions)]
```

(ugly, but that's what is needed to have the actual docstring look good)

Reply:
Uhm, I tested it and it worked on with that indentation. See the attached screenshot (ignore the return type as it was built with also the changes from https://github.com/apache/arrow/pull/11726 in place)
![doc](https://user-images.githubusercontent.com/601423/142453511-725162f3-0c8b-44dd-888c-9f0a1f4e850c.png)

Reply:
In a console you get:
![Screenshot from 2021-11-19 08-04-17](https://user-images.githubusercontent.com/1020496/142579718-7bf56c52-37dd-42ba-a7b8-646422239b1e.png)

Reply:
oic, thanks for catching this, I forgot to check in the console!